game-app/game-core/src/main/java/games/strategy/triplea/ui/panel/move/MovePanel.java

```diff
@@ -261,14 +236,14 @@ private void selectUnitsToMove(
             final List<Unit> unitsToMove = CollectionUtils.getMatches(units, unitsToMoveMatch);
             unitsToMove.sort(UnitComparator.getHighestToLowestMovementComparator());
 
-            final int iterCount = mouseDetails.isAltDown() ? deselectNumber : 1;
+            final int maxCount = mouseDetails.isAltDown() ? deselectNumber : 1;
 
             int addCount = 0;
             for (final Unit unit : unitsToMove) {
               if (!selectedUnits.contains(unit)) {
                 selectedUnits.add(unit);
                 addCount++;
-                if (addCount >= iterCount) {
+                if (addCount >= maxCount) {
                   break;
                 }
               }
```

Comment:
I think this code can be enhanced by rephrasing it similar to this:
```java
final int maxCount = mouseDetails.isAltDown() ? deselectNumber : 1; // deselectNumber is a constant and should be written as DESELECT_NUMBER
final List<Unit> unitsToMove = units.stream()
  .filter(unitsToMoveMatch)
  .filter(Predicate.not(selectedUnits::contains))
  .sorted(UnitComparator.getHighestToLowestMovementComparator())
  .limit(maxCount)
  .collect(Collectors.toList());

selectedUnits.addAll(unitsToMove);
```

Keeps the weird counter out of the way and doesn't sort more than `maxCount` items because AFAIK streams are smart enough to combine limit with sorted.

Also suited to extract this into an extra method

Reply:
I believe it's fine because in this context, but it kinda violates the functional paradigm that there should be no side effects.
Regardless `.forEach` doesn't guarantee any order, `.forEachOrdered` does. Most likely they have the same implementation in this case, but I'd prefer the explicit variant here to make clear order is important

Reply:
Done.